frame/assets/src/functions.rs

```diff
@@ -66,10 +66,11 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 	pub(super) fn new_account(
 		who: &T::AccountId,
 		d: &mut AssetDetails<T::Balance, T::AccountId, DepositBalanceOf<T, I>>,
-		maybe_deposit: Option<DepositBalanceOf<T, I>>,
+		maybe_deposit: Option<(DepositBalanceOf<T, I>, &T::AccountId)>,
 	) -> Result<ExistenceReason<DepositBalanceOf<T, I>>, DispatchError> {
 		let accounts = d.accounts.checked_add(1).ok_or(ArithmeticError::Overflow)?;
-		let reason = if let Some(deposit) = maybe_deposit {
+		let reason = if let Some((deposit, _depositor)) = maybe_deposit {
+			d.deposit = d.deposit.saturating_add(deposit);
```

Comment:
is asset's deposit a sum of account deposits? 
its new here and we do not subtract anything on refund

Reply:
You're right, I don't know why I added that but it is wrong. Removing it.